src/test/scala/sigmastate/CostingSpecification.scala

```diff
@@ -0,0 +1,79 @@
+package sigmastate
+
+import org.ergoplatform.{ErgoLikeContext, ErgoBox}
+import org.ergoplatform.ErgoScriptPredef.TrueProp
+import scorex.crypto.hash.Blake2b256
+import sigmastate.Values.{BooleanConstant, IntConstant, ByteArrayConstant}
+import sigmastate.helpers.{ContextEnrichingTestProvingInterpreter, SigmaTestingCommons}
+import sigmastate.interpreter.ContextExtension
+import sigmastate.interpreter.Interpreter.{ScriptNameProp, ScriptEnv}
+import sigmastate.utxo.CostTable
+import sigmastate.utxo.CostTable._
+
+class CostingSpecification extends SigmaTestingCommons {
+  implicit lazy val IR = new TestingIRContext {
+    override val okPrintEvaluatedEntries = false
+    substFromCostTable = false
+  }
+
+  val env: ScriptEnv = Map(
+    (ScriptNameProp -> s"filename_verify")
+  )
+  val extension: ContextExtension = ContextExtension(Map(
+    1.toByte -> IntConstant(1),
+    2.toByte -> BooleanConstant(true)
+  ))
+  val tokenId = Blake2b256("tokenA")
+  val selfBox = createBox(0, TrueProp, Seq(tokenId -> 10L),
+      Map(ErgoBox.R4 -> ByteArrayConstant(Array[Byte](1, 2, 3)),
+          ErgoBox.R5 -> IntConstant(3)))
+  val context = ErgoLikeContext.dummy(selfBox).withExtension(extension)
+
+  def cost(script: String): Long = {
+    val ergoTree = compiler.compile(env, script)
+    val interpreter = new ContextEnrichingTestProvingInterpreter
+    interpreter.reduceToCrypto(context, env, ergoTree).get._2
+  }
+
+  val ContextVarAccess = accessContextVar + selectField  // `getVar(id)` + `.get`
+  val RegisterAccess = accessRegister + selectField  // `getReg(id)` + `.get`
+  val GTConstCost = comparisonCost + constCost
+
+  property("basic (smoke) tests") {
+
+    cost("{getVar[Boolean](2).get}") shouldBe ContextVarAccess
+
+    cost("{getVar[Int](1).get > 1}") shouldBe (ContextVarAccess + GTConstCost)
```

Comment:
shouldn't it be `(ContextVarAccess + GTConstCost + ConstantDeclarationCost)`?

Reply:
No, GTConstCost already includes constCost.